pkg/metrics/providers/prometheus.go

```diff
@@ -15,8 +15,8 @@ import (
 	flaggerv1 "github.com/weaveworks/flagger/pkg/apis/flagger/v1beta1"
 )
 
-// PrometheusProvider executes promQL queries
-type PrometheusProvider struct {
+// prometheusProvider executes promQL queries
+type prometheusProvider struct {
```

Comment:
I think we should undo this change, the metrics package should expose the providers so it can be used by other projects.